plugins/woocommerce/src/Blocks/BlockTemplatesController.php

```diff
@@ -43,7 +46,37 @@ public function render_woocommerce_template_part( $attributes ) {
 			$template_part = get_block_template( $attributes['theme'] . '//' . $attributes['slug'], 'wp_template_part' );
 
 			if ( $template_part && ! empty( $template_part->content ) ) {
-				return do_blocks( $template_part->content );
+				$content            = do_blocks( $template_part->content );
+				$uncategorized_area = defined( 'WP_TEMPLATE_PART_AREA_UNCATEGORIZED' ) ? \WP_TEMPLATE_PART_AREA_UNCATEGORIZED : 'uncategorized';
+				$area               = $uncategorized_area;
+
+				// Look up area definition.
+				$area_definition = null;
+				$defined_areas   = get_allowed_block_template_part_areas();
+				foreach ( $defined_areas as $defined_area ) {
+					if ( $defined_area['area'] === $area ) {
+						$area_definition = $defined_area;
+						break;
+					}
+				}
+
+				// If $area is not allowed, set it back to the uncategorized default.
+				if ( ! $area_definition ) {
+					$area = $uncategorized_area;
+				}
```

Comment:
Forgive  me if I'm seeing this wrong; I don't see where the $area var gets changed in the loop to need reverting or where does it get used after 🫣

```suggestion
				$area = defined( 'WP_TEMPLATE_PART_AREA_UNCATEGORIZED' ) ? \WP_TEMPLATE_PART_AREA_UNCATEGORIZED : 'uncategorized';

				// Look up area definition.
				$area_definition = null;
				$defined_areas   = get_allowed_block_template_part_areas();
				foreach ( $defined_areas as $defined_area ) {
					if ( $defined_area['area'] === $area ) {
						$area_definition = $defined_area;
						break;
					}
				}
```

Reply:
Wasn't needed. I copied the implementation from core, but we're only using uncategorized which maps to div. https://github.com/woocommerce/woocommerce/pull/51948/commits/e0aa813fadf2efb0a6537a8144f8d876f7b410cf